openedx/core/djangoapps/bookmarks/tests/test_api.py

```diff
@@ -221,9 +221,51 @@ def test_delete_bookmark(self, mock_tracker):
         )
 
         bookmarks_data = api.get_bookmarks(user=self.user)
-        self.assertEqual(len(bookmarks_data), 2)
-        self.assertNotEqual(six.text_type(self.sequential_1.location), bookmarks_data[0]['usage_id'])
-        self.assertNotEqual(six.text_type(self.sequential_1.location), bookmarks_data[1]['usage_id'])
+        self.assertEqual(len(bookmarks_data), 4)
+        self.assertNotEqual(str(self.sequential_1.location), bookmarks_data[0]['usage_id'])
+        self.assertNotEqual(str(self.sequential_1.location), bookmarks_data[1]['usage_id'])
+
+    @patch('openedx.core.djangoapps.bookmarks.api.tracker.emit')
+    def test_delete_bookmarks_with_vertical_block_type(self, mock_tracker):
+        self.assertEqual(len(api.get_bookmarks(user=self.user)), 5)
+
+        api.delete_bookmarks(usage_key=self.vertical_3.location)
+
+        self.assert_bookmark_event_emitted(
+            mock_tracker,
+            event_name='edx.bookmark.removed',
+            course_id=str(self.course_id),
+            bookmark_id=self.bookmark_3.resource_id,
+            component_type=self.vertical_1.location.block_type,
+            component_usage_id=str(self.vertical_3.location),
+        )
+
+        self.assertEqual(len(api.get_bookmarks(self.user)), 4)
+
+    @patch('openedx.core.djangoapps.bookmarks.api.modulestore',
+           return_value=Mock(get_item=Mock(
+               return_value=Mock(get_children=Mock(
+                   return_value=[Mock(get_children=Mock(
+                       return_value=[Mock()]))]))))
+           )
```

Comment:
I could be wrong about this, but wouldn't this patch have worked just as well without specifying all of the return values?  Line 253 would have auto-created all the mocks it needed.

Reply:
> Line 253 would have auto-created all the mocks it needed.

Not exactly - `get_children()` should return a list object (twice).
I've updated the mocks chain setup by moving valuable code from the decoration section.